guides/source/testing.md

```diff
@@ -1685,121 +1761,88 @@ end
-
-### Additional View-Based Assertions
-
-There are more assertions that are primarily used in testing views:
-
-| Assertion                                                 | Purpose |
-| --------------------------------------------------------- | ------- |
-| `assert_select_email`                                     | Allows you to make assertions on the body of an e-mail. |
-| `assert_select_encoded`                                   | Allows you to make assertions on encoded HTML. It does this by un-encoding the contents of each element and then calling the block with all the un-encoded elements.|
-| `css_select(selector)` or `css_select(element, selector)` | Returns an array of all the elements selected by the _selector_. In the second variant it first matches the base _element_ and tries to match the _selector_ expression on any of its children. If there are no matches both variants return an empty array.|
-
-Here's an example of using `assert_select_email`:
-
-```ruby
-assert_select_email do
-  assert_select "small", "Please click the 'Unsubscribe' link if you want to opt-out."
+assert_dom "ol" do
+  assert_dom "li", 8
 end
 ```
 
-Testing View Partials
----------------------
-
-Partial templates - usually called "partials" - are another device for breaking the rendering process into more manageable chunks. With partials, you can extract pieces of code from your templates to separate files and reuse them throughout your templates.
-
-View tests provide an opportunity to test that partials render content the way you expect. View partial tests reside in `test/views/` and inherit from `ActionView::TestCase`.
-
-To render a partial, call `render` like you would in a template. The content is
-available through the test-local `#rendered` method:
+The `assert_dom_equal` method tests two HTML strings for equivalency, as in the
+example below:
 
 ```ruby
-class ArticlePartialTest < ActionView::TestCase
-  test "renders a link to itself" do
-    article = Article.create! title: "Hello, world"
-
-    render "articles/article", article: article
-
-    assert_includes rendered, article.title
-  end
-end
+assert_dom_equal '<a href="http://www.further-reading.com">Read more</a>',
+  link_to("Read more", "http://www.further-reading.com")
 ```
 
-Tests that inherit from `ActionView::TestCase` also have access to [`assert_select`](#testing-views) and the [other additional view-based assertions](#additional-view-based-assertions) provided by [rails-dom-testing][]:
-
-```ruby
-test "renders a link to itself" do
-  article = Article.create! title: "Hello, world"
-
-  render "articles/article", article: article
-
-  assert_select "a[href=?]", article_url(article), text: article.title
-end
-```
+These assertions are quite powerful. For more advanced usage, refer to their
```

Comment:
I think we already mentioned that these assertions are powerful above, so maybe simply this:
```suggestion
For more advanced usage, refer to the
```